### DISCUSSION TITLE
please give some suggestions for glowtts config. I Change something or keep them default? I'm using hindi language dataset

### DISCUSSION THREAD

**Discussion by omkarade** [coqui-ai/TTS#1990] [General Q&A]
class TTS.tts.configs.glow_tts_config.GlowTTSConfig(output_path='output', logger_uri=None, run_name='run', project_name=None, run_description='🐸Coqui trainer run.', print_step=25, plot_step=100, model_param_stats=False, wandb_entity=None, dashboard_logger='tensorboard', log_model_step=None, save_step=10000, save_n_checkpoints=5, save_checkpoints=True, save_all_best=False, save_best_after=10000, target_loss=None, print_eval=False, test_delay_epochs=0, run_eval=True, run_eval_steps=None, distributed_backend='nccl', distributed_url='tcp://localhost:54321', mixed_precision=False, epochs=1000, batch_size=32, eval_batch_size=16, grad_clip=5.0, scheduler_after_epoch=True, lr=0.001, optimizer='RAdam', optimizer_params=<factory>, lr_scheduler='NoamLR', lr_scheduler_params=<factory>, use_grad_scaler=False, cudnn_enable=True, cudnn_deterministic=False, cudnn_benchmark=False, training_seed=54321, model='glow_tts', num_loader_workers=0, num_eval_loader_workers=0, use_noise_augment=False, audio=<factory>, use_phonemes=False, phonemizer=None, phoneme_language=None, compute_input_seq_cache=False, text_cleaner=None, enable_eos_bos_chars=False, test_sentences_file='', phoneme_cache_path=None, characters=None, add_blank=False, batch_group_size=0, loss_masking=None, min_audio_len=1, max_audio_len=inf, min_text_len=1, max_text_len=inf, compute_f0=False, compute_linear_spec=False, precompute_num_workers=0, start_by_longest=False, datasets=<factory>, test_sentences=<factory>, eval_split_max_size=None, eval_split_size=0.01, use_speaker_weighted_sampler=False, speaker_weighted_sampler_alpha=1.0, use_language_weighted_sampler=False, language_weighted_sampler_alpha=1.0, use_length_weighted_sampler=False, length_weighted_sampler_alpha=1.0, num_chars=None, encoder_type='rel_pos_transformer', encoder_params=<factory>, use_encoder_prenet=True, hidden_channels_enc=192, hidden_channels_dec=192, hidden_channels_dp=256, dropout_p_dp=0.1, dropout_p_dec=0.05, mean_only=True, out_channels=80, num_flow_blocks_dec=12, inference_noise_scale=0.0, kernel_size_dec=5, dilation_rate=1, num_block_layers=4, num_speakers=0, c_in_channels=0, num_splits=4, num_squeeze=2, sigmoid_scale=False, d_vector_dim=0, data_dep_init_steps=10, style_wav_for_test=None, length_scale=1.0, use_speaker_embedding=False, speakers_file=None, use_d_vector_file=False, d_vector_file=False, min_seq_len=3, max_seq_len=500, r=1)

**Comment by poynty** [comment]
I can't help you I am afraid, but I can make that wall of text slightly less headache inducing so perhaps someone else can cast an eye on it.


```
class TTS.tts.configs.glow_tts_config.GlowTTSConfig(
  output_path = 'output',
  logger_uri = None,
  run_name = 'run',
  project_name = None,
  run_description = '🐸Coqui trainer run.',
  print_step = 25,
  plot_step = 100,
  model_param_stats = False,
  wandb_entity = None,
  dashboard_logger = 'tensorboard',
  log_model_step = None,
  save_step = 10000,
  save_n_checkpoints = 5,
  save_checkpoints = True,
  save_all_best = False,
  save_best_after = 10000,
  target_loss = None,
  print_eval = False,
  test_delay_epochs = 0,
  run_eval = True,
  run_eval_steps = None,
  distributed_backend = 'nccl',
  distributed_url = 'tcp://localhost:54321',
  mixed_precision = False,
  epochs = 1000,
  batch_size = 32,
  eval_batch_size = 16,
  grad_clip = 5.0,
  scheduler_after_epoch = True,
  lr = 0.001,
  optimizer = 'RAdam',
  optimizer_params =,
  lr_scheduler = 'NoamLR',
  lr_scheduler_params =,
  use_grad_scaler = False,
  cudnn_enable = True,
  cudnn_deterministic = False,
  cudnn_benchmark = False,
  training_seed = 54321,
  model = 'glow_tts',
  num_loader_workers = 0,
  num_eval_loader_workers = 0,
  use_noise_augment = False,
  audio =,
  use_phonemes = False,
  phonemizer = None,
  phoneme_language = None,
  compute_input_seq_cache = False,
  text_cleaner = None,
  enable_eos_bos_chars = False,
  test_sentences_file ='',
  phoneme_cache_path = None,
  characters = None,
  add_blank = False,
  batch_group_size = 0,
  loss_masking = None,
  min_audio_len = 1,
  max_audio_len = inf,
  min_text_len = 1,
  max_text_len = inf,
  compute_f0 = False,
  compute_linear_spec = False,
  precompute_num_workers = 0,
  start_by_longest = False,
  datasets =,
  test_sentences =,
  eval_split_max_size = None,
  eval_split_size = 0.01,
  use_speaker_weighted_sampler = False,
  speaker_weighted_sampler_alpha = 1.0,
  use_language_weighted_sampler = False,
  language_weighted_sampler_alpha = 1.0,
  use_length_weighted_sampler = False,
  length_weighted_sampler_alpha = 1.0,
  num_chars = None,
  encoder_type = 'rel_pos_transformer',
  encoder_params =,
  use_encoder_prenet = True,
  hidden_channels_enc = 192,
  hidden_channels_dec = 192,
  hidden_channels_dp = 256,
  dropout_p_dp = 0.1,
  dropout_p_dec = 0.05,
  mean_only = True,
  out_channels = 80,
  num_flow_blocks_dec = 12,
  inference_noise_scale = 0.0,
  kernel_size_dec = 5,
  dilation_rate = 1,
  num_block_layers = 4,
  num_speakers = 0,
  c_in_channels = 0,
  num_splits = 4,
  num_squeeze = 2,
  sigmoid_scale = False,
  d_vector_dim = 0,
  data_dep_init_steps = 10,
  style_wav_for_test = None,
  length_scale = 1.0,
  use_speaker_embedding = False,
  speakers_file = None,
  use_d_vector_file = False,
  d_vector_file = False,
  min_seq_len = 3,
  max_seq_len = 500,
  r = 1)
```
